tests/signal/test_spectral.pyi

```diff
@@ -0,0 +1,65 @@
+from typing import Any, TypeAlias
+from typing_extensions import assert_type
+
+import numpy as np
+import numpy.typing as npt
+from scipy.signal import istft, spectrogram
+
+_Array_f8: TypeAlias = npt.NDArray[np.float64]
+_ArrayReal: TypeAlias = npt.NDArray[np.floating[Any]]
+_ArrayComplex: TypeAlias = npt.NDArray[np.complexfloating[Any, Any]]
+
+# test spectrogram function overloads
+assert_type(spectrogram(np.linspace(200, 300, 256)), tuple[_Array_f8, _Array_f8, _ArrayReal])
+assert_type(spectrogram(np.linspace(200, 300, 256), mode="psd"), tuple[_Array_f8, _Array_f8, _ArrayReal])
+assert_type(spectrogram(np.linspace(200, 300, 256), mode="magnitude"), tuple[_Array_f8, _Array_f8, _ArrayReal])
+assert_type(spectrogram(np.linspace(200, 300, 256), mode="angle"), tuple[_Array_f8, _Array_f8, _ArrayReal])
+assert_type(spectrogram(np.linspace(200, 300, 256), mode="phase"), tuple[_Array_f8, _Array_f8, _ArrayReal])
```

Comment:
You could test these different `mode` values in a single call by extracting the values as a "gradual type" like

```
spectrogram_mode_real: Literal["psd", "magnitude", ...]
```

So that's the main advantage of using `.pyi` for type-testing, in case you were wondering :)